docs/additional-resources/openms-git-workflow.md

```diff
@@ -0,0 +1,174 @@
+* Name the OpenMS remote `upstream` and always push directly to `origin` (`git push origin <branch-name>`).
+* When updating your fork, consider using `git fetch upstream` followed by `git merge --ff-only upstream/develop` to avoid creating merge commits in `develop`.
+  If you never commit to `develop` this should always succeed and (if a commit accidentally went to develop) warn you instead of creating a merge commit.
+
+## Create New Feature
+
+All features start from `develop`.
+
+```bash
+$ git checkout develop
+$ git checkout -b feature/your-cool-new-feature
+```
+All commits related to this feature will then go into the branch `feature/your-cool-new-feature`.
+
+## Keeping your Feature Branch in Sync with Develop Branch
+
+While working on your feature branch, it is usual that development continues and new features get integrated into the main development branch. This means your feature branch lags behind `develop`. To get your feature branch up-to-date, rebase your feature branch on `develop` using:
+
+```bash
+$ git checkout feature/myfeaturebranch
+$ git rebase develop
+```
+
+The above commands:
+
+1. Performs a rewind of your commits until the branching point.
+2. Applies all commits that have been integrated into `develop`.
+3. Reapplies your commits on top of the commits integrated into `develop`.
+
+For more information, refer to a [visual explanation of rebasing](http://git-scm.com/book/en/v2/Git-Branching-Rebasing).
+
+> **_NOTE:_** Do not rebase published branches (e.g. branches that are part of a pull request). If you created a pull request, you should only add commits in your feature branch to fix things that have been discussed. After your pull request contains all fixes, you are ready to merge the pull request into develop without rebasing (see e.g. rebase-vs-merge).
+
+## Finish a Feature
+
+Features that should go into the main development line of OpenMS should be integrated via a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests). This allows the development community of OpenMS to discuss the changes and suggest possible improvements.
+
+After opening the pull request via the GitHub web site, GitHub will try to create the pull request against the branch that you branched off from. Please check the branch that you are opening the pull request against before submitting the pull request. If any changes are made, a new pull request is required. Select **Allow others to make changes to this pull request** so that maintainers can directly help to solve problems.
+
+Open pull requests only after checking code-style, documentation and passing tests. Pull requests that do not pass CI or code review will not be merged until the problems are solved. It is recommended that you read the [checklist](pull-request-checklist.md) before you submit a pull request.
```

Comment:
let's replace:

> checklist
w/
pull request guidelines.

(skip changing the file name)